WebRoot/js/zimbraMail/prefs/model/ZmFilterRules.js

```diff
@@ -393,22 +393,21 @@ function(index, notify, callback, result) {
 ZmFilterRules.prototype._handleErrorSaveRules =
 function(ex) {
 	if (ex.code == ZmCsfeException.SVC_PARSE_ERROR ||
-		ex.code == ZmCsfeException.SVC_INVALID_REQUEST ||
+		ex.code == ZmCsfeException.SIEVE_SCRIPT_MAX_SIZE_EXCEPTION ||
 		ex.code == ZmCsfeException.MAIL_NO_SUCH_FOLDER ||
 		ex.code == ZmCsfeException.MAIL_NO_SUCH_TAG)
 	{
 		var msgDialog = appCtxt.getMsgDialog();
 
 		var msg = '';
-		switch(ex.code) {
-			case ZmCsfeException.MAIL_NO_SUCH_FOLDER:
-				msg = ZmMsg.errorNoSuchFolder;
-				break;
-			case ZmCsfeException.MAIL_NO_SUCH_TAG:
-				msg = ZmMsg.errorNoSuchTag;
-				break;
-			default:
-				msg = [ZmMsg.filterError, " ", AjxStringUtil.htmlEncode(ex.msg)].join("");
+		if (ex.code === ZmCsfeException.MAIL_NO_SUCH_FOLDER) {
```

Comment:
not sure if we want to change switch statement to if/else now

Reply:
corrected..